configure

```diff
@@ -257,33 +257,54 @@ touch s.h m.h Makefile
 
 # Write options to Makefile
 
+config() {
+# This function hardens the generated Makefile against '#' symbols
+# present in a source path (opam-compiler-conf may pick such directory
+# names if working from a branch named 'PR#4242-answer-all-questions')
+# by escaping them into '\#'.
+
+# When injecting data in Makefiles, it is customary to also escape
+# '$', which get turned into '$$'. However, this transformation is
+# invalid here as some of the variables are meant to be code
+# interpreted by make: for example, passing
+#   --bindir "$(PREFIX)/bin2"
+# is explicitly supported (see "or relative to $(PREFIX)" messages above).
+
+# Finally, it is also impossible for the user to escape the '#' signs
+# before calling this configure script, given that
+# $(PREFIX) is also injected in C code where this escape is invalid
+# -- see the definition of the OCAML_STDLIB_DIR macro below.
+
+  echo $1=$2 | sed 's/\#/\\#/g' >> Makefile
```

Comment:
So I tried to use `echo $1=\"$2\"` here but I don't like the fact that it makes the result very different from the previous `config/Makefile` where variables are not quoted. In the next iteration (I'll push now), I used `"$partialld -o\\ \$(EMPTY)"` instead as some other variables were already doing, does that sound reasonable to you?

If I diff the config/Makefile before and after the change, the differences are:

- `CPPFLAGS` had a trailing space that is now gone (I think this is correct, it comes from the space before the empty variable `internal_cppflags`)
- `BYTECCLIBS` has 18 trailing spaces that are now gone (that again seems correct)
- `NATIVECCLIBS` had two spaces instead of one in the middle
- and `PACKLD` that needs a fix as you pointed out

Sorry, I should have made this comparison earlier.
-